src/common/lib/util/utils.ts

```diff
@@ -547,3 +547,20 @@ export function shallowEquals(source: Record<string, unknown>, target: Record<st
     Object.keys(target).every((key) => target[key] === source[key])
   );
 }
+
+export function matchDerivedChannel(name: string) {
+  /**
+   * This warning is triggered because the RegExp uses nested quantifiers,
+   * but it does not create any situation where the regex engine has to
+   * explore a large number of possible matches so it’s safe to ignore
+   */
+  const regex = /^(\[([^?]*)(?:(.*))\])?(.+)$/; // eslint-disable-line
+  const match = name.match(regex)
+  if (!match || !match.length || match.length < 5) {
+    throw new Error(`regex match failed`);
+  }
+  if (match[2]) {
+    throw new Error(`cannot use a derived option with a ${match[2]} channel`);
```

Comment:
Might be worth adding some comments around the logic in this function - it's quite hard to read the regex and work out exactly what causes this error message